### PR TITLE
[MM-37423] Custom Status: Don't show 'Custom Date and Time' for recent statuses

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
Custom Status: Don't show 'Custom Date and Time' for recent statuses

#### Ticket Link
Fixes [https://mattermost.atlassian.net/browse/MM-37423](https://mattermost.atlassian.net/browse/MM-37423)

#### Checklist
- [x] Has UI changes

#### Device Information
This PR was tested on: 
Simulator IPhone 12 - Version: 14.5,
Emulator Google Pixel XL API 29 - Version:10

#### Screenshots
Simulator Screen Shot - iPhone 12 - 2021-08-24 at 12 55 06

#### Release Note
```release-note
* Remove Date and Time label from recents custom status
```

### CODE REVIEW

File: app/screens/custom_status/custom_status_suggestion.tsx [hunk 1]
```diff
@@ -74,7 +74,9 @@ const CustomStatusSuggestion = ({handleSuggestionClick, emoji, text, theme, sepa
                                 textStyle={style.customStatusText}
                             />
                         </View>
-                        {Boolean(duration && isExpirySupported) && (
+                        {Boolean(duration &&
+                            duration !== CustomStatusDuration.DATE_AND_TIME &&
```
Comment (michelengelen) [inline app/screens/custom_status/custom_status_suggestion.tsx:78]: what about `CustomStatusDuration.CUSTOM_DATE_TIME`?

**Review by michelengelen — CHANGES_REQUESTED** [review]

File: app/screens/custom_status/custom_status_suggestion.tsx (hunk 1 above)
Comment (AshishDhama) [reply app/screens/custom_status/custom_status_suggestion.tsx:78]: @michelengelen Mobile don't have CUSTOM_DATE_TIME 
```
export enum CustomStatusDuration {
    DONT_CLEAR = '',
    THIRTY_MINUTES = 'thirty_minutes',
    ONE_HOUR = 'one_hour',
    FOUR_HOURS = 'four_hours',
    TODAY = 'today',
    THIS_WEEK = 'this_week',
    DATE_AND_TIME = 'date_and_time',
}
```

File: app/screens/custom_status/custom_status_suggestion.tsx [hunk 2]
```diff
@@ -74,7 +74,9 @@ const CustomStatusSuggestion = ({handleSuggestionClick, emoji, text, theme, sepa
                                 textStyle={style.customStatusText}
                             />
                         </View>
-                        {Boolean(duration && isExpirySupported) && (
+                        {Boolean(duration &&
```
Comment (anurag6713) [inline app/screens/custom_status/custom_status_suggestion.tsx:77]: Small change :) Can we move the condition out of JSX and use the variable for readability? Thank you!
```
const showCustomStatus = Boolean(duration && duration !== CustomStatusDuration.DATE_AND_TIME && isExpirySupported);
```

Comment (AshishDhama) [reply app/screens/custom_status/custom_status_suggestion.tsx:77]: Sure

**Review by matthewbirtch — APPROVED** [review]
Seems to work as expected @AshishDhama. I verified that the expiry time does not show for recent statuses if a custom date/time was chosen. Thanks!

**Review by anurag6713 — APPROVED** [review]
LGTM! 🎉 Thank you.

**Review by michelengelen — APPROVED** [review]
LGTM

**Review by jgilliam17 — APPROVED** [review]
Thank you @AshishDhama 
Tested, looks good to merge.
- Verified "Date and Time" text is not shown for recent statuses that had custom date/time set.

**Pull request merged**